velox/common/memory/MappedMemory.h

```diff
@@ -190,13 +250,36 @@ class MappedMemory {
   // Returns the number of freed bytes.
   virtual int64_t free(Allocation& allocation) = 0;
 
+  // Makes a contiguous mmap of 'numPages'. Advises away the required
+  // number of free pages so as not to have resident size exceed the
+  // capacity if capacity is bounded. Returns false if sufficient free
+  // pages do not exist. 'collateral' and 'allocation' are freed and
+  // unmapped or advised away to provide pages to back the new
+  // 'allocation'. This will always succeed if collateral and
+  // allocation together cover the new size of
+  // allocation. 'allocation' is newly mapped and hence zeroed. The
+  // contents of 'allocation' and 'collateral' are freed in all cases,
+  // also if the allocation fails. 'beforeAllocCB can be used to
+  // update trackers. It may throw and the end state will be
+  // consistent, with no new allocation and 'allocation' and
+  // 'collateral' cleared.
+  virtual bool allocateContiguous(
+      MachinePageCount numPages,
+      Allocation* FOLLY_NULLABLE collateral,
+      ContiguousAllocation& allocation,
+      std::function<void(int64_t)> beforeAllocCB = nullptr) = 0;
+
+  virtual void freeContiguous(ContiguousAllocation& allocation) = 0;
+
   // Checks internal consistency of allocation data
   // structures. Returns true if OK.
   virtual bool checkConsistency() = 0;
 
   static void destroyTestOnly();
 
-  virtual const std::vector<MachinePageCount>& sizes() const = 0;
+  virtual const std::vector<MachinePageCount>& sizes() const {
```

Comment:
nit: perhaps, this would be clearer as sizeClasses, e.g. sizeClass specifies number of pages.